File: 2095-minimum-number-of-swaps-to-make-the-string-balanced/minimum-number-of-swaps-to-make-the-string-balanced.go
func minSwaps(s string) int {
    arr:= []byte{}
    n:= len(s)
    Imbalance:=0
    for i:=0;i<n;i++{
        c:=s[i]
        if c=='['{
            arr=append(arr,c)
        }else if len(arr)>0{
            arr = arr[:len(arr)-1]
        }else{
            Imbalance++;
        }
    }
    return (Imbalance+1)/2;
}

File: 2095-minimum-number-of-swaps-to-make-the-string-balanced/minimum-number-of-swaps-to-make-the-string-balanced.cpp
class Solution {
public:
    int minSwaps(string s) {
        stack<char>st;
        int n = s.size();
        int imbalanceCount = 0;

        for (int i=0;i<n;i++) {
            char c = s[i];
            if (c == '[') {
                st.push(c);
            } else {
                if(!st.empty() && st.top()=='['){
                    st.pop();
                }else {
                    imbalanceCount++;
                }
            }

        }

        return (imbalanceCount+1)/2;
    }
};
